### PR TITLE
feat: Progress snackbar

### PR DESCRIPTION
**Pull request by ChrisBaidoo**
## Description

Move Progress Snackbar to STYLE

https://user-images.githubusercontent.com/37402968/155290723-930a7e4a-13d2-4c2a-a2da-b1fafa97fed8.mov

## Checklist:

Put an `x` in the boxes that apply to this pull request (you can also fill these out after opening the pull request). If you're unsure about any of these, don't hesitate to leave a comment on this pull request!

- [ ] I have read the gliff.ai Contribution Guide.
- [ ] I have requested to **pull a branch** and not from main.
- [ ] I have checked all commit message styles match the requested structure.
- [ ] My code follows the style guidelines of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have performed a self-review of my own code.
- [ ] I have assigned **3 or less** reviewers.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] My changes generate no new warnings.
- [ ] I have made corresponding changes to the documentation.
- [ ] New database changes have been committed.
- [ ] If appropriate, I have bumped any version numbers.

### CODE REVIEW

File: src/ProgressSnackbar.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+} from "@mui/material";
+// import makeStyles from "@mui/styles/makeStyles";
+import { theme } from "./theme";
+
+export const imgSrc = (src: string, type = "svg"): string =>
+  new URL(`/src/assets/icons/${src}.${type}`, import.meta.url).href;
+
+declare module "@mui/styles/defaultTheme" {
+  // eslint-disable-next-line @typescript-eslint/no-empty-interface
+  interface DefaultTheme extends Theme {}
```
Comment (cooper667) [inline src/ProgressSnackbar.tsx:22]: We should be able to put this somewhere global rather than in every file?

Comment (ChrisBaidoo) [reply src/ProgressSnackbar.tsx:22]: I have tried doing this in the theme.tsx. Where I'm I going wrong?  
This is a syntax I am unfamiliar with.  @cooper667 
Screenshot 2022-02-22 at 12 06 34

Comment (cooper667) [reply src/ProgressSnackbar.tsx:22]: try creating `theme.d.ts`

and putting

```
declare module "@mui/styles/defaultTheme" {
  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  interface DefaultTheme extends Theme {}
 }
 ```
  
  in it

Comment (cooper667) [reply src/ProgressSnackbar.tsx]: Yea we're limited to 3 and are already using those on a bunch of dependabot branches for some reason

**Review by cooper667 — DISMISSED** [review]

**Review by cooper667 — APPROVED** [review]

**Review by joshuajames-smith — APPROVED** [review]

**Pull request merged**